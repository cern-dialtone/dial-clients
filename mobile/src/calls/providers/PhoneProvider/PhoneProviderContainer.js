import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import {
  connectionActions,
  callActions,
  recentCallsActions,
  authActions,
  authActionFactory
} from 'dial-core';
import { PhoneProvider } from './PhoneProvider';
import { phoneService } from './PhoneService';

export function mapStateToProps({ calls, auth }) {
  return {
    authToken: auth.authToken,
    doNotDisturb: calls.status.doNotDisturb,
    call: calls.call,
    toneToken: auth.toneToken
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      ...authActions,
      logout: authActionFactory(process.env.REACT_APP_API_ENDPOINT).logout,
      ...connectionActions,
      ...callActions,
      ...recentCallsActions
    },
    dispatch
  );
}

export default phoneService(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(PhoneProvider)
);
