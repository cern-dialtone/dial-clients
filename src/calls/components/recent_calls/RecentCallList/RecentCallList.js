import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { Item } from 'semantic-ui-react';
import RecentCall from 'calls/components/recent_calls/RecentCall/RecentCall';
import ScrollableContent from 'common/components/ScrollableContent/ScrollableContent';
import RecentCallModal from 'calls/components/recent_calls/RecentCallModal/RecentCallModal';
import { logMessage } from 'common/utils/logs';

/**
 * Displays a scrollable list of RecentCall Components
 */
function RecentCallList({ recentCalls }) {
  const [open, setIsOpen] = useState(false);
  const [selectedCall, setSelectedCall] = useState(null);

  const selectRecentCall = recentCall => {
    setIsOpen(true);
    setSelectedCall(recentCall);
  };

  const handleClose = () => {
    logMessage('handle close');
    setIsOpen(false);
    setSelectedCall(null);
  };

  return (
    <ScrollableContent>
      <Item.Group link data-testid="RecentCallList">
        {recentCalls.map((item, index) => {
          if (item && !item.name) return null;
          return (
            <RecentCall
              key={`recent-${index.toString()}`}
              recentCallId={`recent-${index.toString()}`}
              recentCall={item}
              onClick={() => selectRecentCall(item)}
            />
          );
        })}
      </Item.Group>
      <RecentCallModal
        selectedCall={selectedCall}
        open={open}
        handleClose={handleClose}
      />
    </ScrollableContent>
  );
}

RecentCallList.propTypes = {
  recentCalls: PropTypes.arrayOf(
    PropTypes.shape({
      item: PropTypes.shape({
        name: PropTypes.string.isRequired
      })
    })
  )
};

RecentCallList.defaultProps = {
  recentCalls: []
};

export default RecentCallList;
