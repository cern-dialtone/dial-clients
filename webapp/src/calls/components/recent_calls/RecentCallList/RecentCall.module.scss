.ItemHeader{
  font-size: 1em !important;
}

.Content{
  margin-top: 0 !important;
}

.ExtraContent{
  margin-top: 0 !important;
}