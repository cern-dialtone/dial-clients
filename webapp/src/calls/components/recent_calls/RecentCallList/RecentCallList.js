import React, { Component } from "react";
import PropTypes from "prop-types";

import { Item } from "semantic-ui-react";
import RecentCall from "calls/components/recent_calls/RecentCall/RecentCall";
import ScrollableContent from "common/components/ScrollableContent/ScrollableContent";

/**
 * Displays a scrollable list of RecentCall Components
 */
class RecentCallList extends Component {
  static propTypes = {
    recentCalls: PropTypes.array.isRequired
  };

  render() {
    const { recentCalls } = this.props;

    return (
      <ScrollableContent>
        <Item.Group link>
          {recentCalls.map((item, index) => {
            return <RecentCall key={`recent-${index}`} recentCall={item} />;
          })}
        </Item.Group>
      </ScrollableContent>
    );
  }
}

export default RecentCallList;
