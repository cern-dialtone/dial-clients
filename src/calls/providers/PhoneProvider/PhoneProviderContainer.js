import { connect } from 'react-redux';

import { bindActionCreators } from 'redux';

import {
  callActions,
  connectionActions,
  recentCallsActions,
  authActions
} from 'dial-core';

import { info, success, warning } from 'common/actions/notifications';
import PhoneProvider from 'calls/providers/PhoneProvider/PhoneProvider';
import { withPhoneService } from 'calls/providers/PhoneProvider/PhoneService';

import dialBackendApi from 'services/api';

export function mapStateToProps({ auth, call, status, connection }) {
  return {
    doNotDisturb: status.doNotDisturb,
    call,
    tempRemote: call.tempRemote,
    authToken: auth.authToken,
    connected: connection.connected
  };
}

export function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      ...authActions,
      logout: dialBackendApi().logout,
      ...connectionActions,
      ...callActions,
      ...recentCallsActions,
      success,
      info,
      warning
    },
    dispatch
  );
}

export default withPhoneService(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(PhoneProvider)
);
