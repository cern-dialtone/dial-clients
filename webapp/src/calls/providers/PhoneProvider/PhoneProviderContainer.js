import { connect } from "react-redux";

import { bindActionCreators } from "redux";
import * as connectionActionCreators from "calls/actions/connection";
import * as callActionCreators from "calls/actions/call";
import * as recentActionCreators from "calls/actions/recent";
import * as authActionCreators from "auth/actions/auth";
import { info, success, warning } from "common/actions/notifications";
import { phoneService } from "calls/providers/PhoneProvider/PhoneProvider";
import PhoneProvider from "calls/providers/PhoneProvider/PhoneProvider";

export function mapStateToProps({ calls, auth }) {
  return {
    recipient: calls.call ? calls.call.recipient : undefined,
    onCall: calls.call ? calls.call.onCall : false,
    token: auth.token,
    doNotDisturb: calls.status.doNotDisturb
  };
}

export function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      ...connectionActionCreators,
      ...callActionCreators,
      ...recentActionCreators,
      ...authActionCreators,
      success,
      info,
      warning
    },
    dispatch
  );
}

export default phoneService(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(PhoneProvider)
);
