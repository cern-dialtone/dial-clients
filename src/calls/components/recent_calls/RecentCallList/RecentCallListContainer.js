import { connect } from 'react-redux';
import RecentCallList from './RecentCallList';

function mapStateToProps({ recent }) {
  return {
    recentCalls: recent.recentCalls
  };
}

export default connect(mapStateToProps)(RecentCallList);
