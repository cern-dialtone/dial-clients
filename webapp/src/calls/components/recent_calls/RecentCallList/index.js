import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import RecentCallList from "./RecentCallList";

function mapStateToProps({ calls }) {
  return {
    recentCalls: calls.recent.recentCalls
  };
}

export default withRouter(
  connect(
    mapStateToProps
  )(RecentCallList)
);
