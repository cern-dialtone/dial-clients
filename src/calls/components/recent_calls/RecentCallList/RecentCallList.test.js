import React from 'react';
import { render, waitForElement, fireEvent } from '@testing-library/react';

import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

import { createHashHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';

import RecentCallList from './RecentCallList';
import PhoneProviderContainer from 'calls/providers/PhoneProvider/PhoneProviderContainer';
import { Provider } from 'react-redux';

global.remote = {
  require: jest.fn()
};

describe('RecentCallList tests', () => {
  it('renders empty List without any errors', async () => {
    const { asFragment, getByTestId } = render(<RecentCallList />);
    await waitForElement(() => getByTestId('RecentCallList'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a List without any errors', async () => {
    const recentCalls = [
      {
        name: '12345',
        endTime: 12345,
        phoneNumber: '222333',
        missed: true,
        incoming: true
      },
      {
        name: '123456',
        endTime: 12345,
        phoneNumber: '444555',
        missed: false,
        incoming: false
      }
    ];
    const { asFragment, getByTestId } = render(
      <RecentCallList recentCalls={recentCalls} />
    );
    await waitForElement(() => getByTestId('RecentCallList'));
    await waitForElement(() => getByTestId('recent-0'));
    await waitForElement(() => getByTestId('recent-1'));
  });

  it('handles the recent call click and modal close', async () => {
    const recentCalls = [
      {
        name: '12345',
        endTime: 12345,
        phoneNumber: '222333',
        missed: true,
        incoming: true
      },
      {
        name: '123456',
        endTime: 12345,
        phoneNumber: '444555',
        missed: false,
        incoming: false
      }
    ];

    const history = createHashHistory();
    const middlewares = [thunk, routerMiddleware(history)];
    const mockStore = configureMockStore(middlewares);

    const initialState = {
      status: {
        doNotDisturb: false
      },
      call: {
        tempRemote: null
      },
      auth: {
        authToken: null
      },
      settings: {
        devices: []
      },
      connection: {
        connected: true
      }
    };
    const store = mockStore(initialState);
    store.dispatch = jest.fn();

    const { getByTestId, queryByTestId, container, getByLabelText } = render(
      <Provider store={store}>
        <PhoneProviderContainer>
          <RecentCallList recentCalls={recentCalls} />
        </PhoneProviderContainer>
      </Provider>
    );
    await waitForElement(() => getByTestId('RecentCallList'));
    await waitForElement(() => getByTestId('recent-0'));
    await waitForElement(() => getByTestId('recent-1'));

    expect(queryByTestId('RecentCallModal')).toBeNull();
    // TODO Test this. Currently it's hard since it expects UserPhoneNumberButton to have a phoneService
    // Unfortunatly, the imports are provoking an error.
    const firstRecentCall = await getByTestId('recent-0');
    fireEvent.click(firstRecentCall);
    await waitForElement(() => getByTestId('RecentCallModal'));
    const closeButton = getByLabelText('Close call details modal');

    fireEvent.click(closeButton);
    expect(queryByTestId('RecentCallModal')).toBeNull();
  });
});
