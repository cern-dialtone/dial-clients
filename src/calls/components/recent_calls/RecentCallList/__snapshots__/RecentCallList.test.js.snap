// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentCallList tests renders empty List without any errors 1`] = `
<DocumentFragment>
  <div
    class="ScrollableContent"
  >
    <div
      class="ui link items"
      data-testid="RecentCallList"
    />
  </div>
</DocumentFragment>
`;
